test/e2e/deployment-plans/interior.go

```diff
@@ -127,6 +136,43 @@ func testInteriorScaleUp(f *framework.Framework) {
 	By("Verifying the Deployment has reached 4 replicas")
 	err = framework.WaitForDeployment(f.KubeClient, f.Namespace, "interior-interconnect", 4, framework.RetryInterval, framework.Timeout)
 	Expect(err).NotTo(HaveOccurred())
+
+	By("Verifying the Deployment contains 4 pods")
+	pods, err := f.ListPodsForDeployment(dep)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(len(pods.Items)).To(Equal(4))
+
+	By("Verifying the Network contains 4 nodes on each pod")
```

Comment:
In anticipation of this check being used in other tests, could we introduce `f.ListNodesForPod(pod)` either by adding to the deployment.go in the framework or creating a pod.go as we are likely to have more pod level checks. Also, Nodes might be an overloaded term, should it be NodeEntities or something similar?

Reply:
I agree we must have a common component for such validations, as it will enable reuse by other tests. But I am still wondering if the provider of such method should really be the framework or pod (.go) files.
That's because this is not directly a framework or pod related action, but a qdr network related thing, that depends on a pod/framework. So what if we have a provider named as: qdr-status, qdr-network, or something like it, to ensure the "ListNodesForPod" or other qdr network related methods are available in the same provider.